src/Voron/Impl/Transaction.cs

```diff
@@ -213,21 +233,20 @@ public Table OpenTable(TableSchema schema, string name)
 
         public Table OpenTable(TableSchema schema, Slice name)
         {
-            _tables ??= new Dictionary<Slice, Table>(SliceStructComparer.Instance);
-
-            if (_tables.TryGetValue(name, out Table value))
-                return value;
+            _tables ??= new Dictionary<TableKey, Table>();
 
             var clonedName = name.Clone(Allocator);
+            var key = new TableKey(clonedName, schema.Compressed);
```

Comment:
You shouldn't clone the name just for calling `TryGetValue()`. You must clone only if you do `_tables[key] = new TableKey(clonedName, schema.Compressed);`

Reply:
In general we want to avoid doing the clone on each attempt to get a table